static/app/components/core/alert.chonk.tsx

```diff
@@ -0,0 +1,126 @@
+import {css, type DO_NOT_USE_ChonkTheme} from '@emotion/react';
+
+import type {AlertProps} from 'sentry/components/alert';
+import type {ChonkPropMapping} from 'sentry/utils/theme/withChonk';
+
+const chonkAlertPropMapping: ChonkPropMapping<AlertProps, ChonkAlertProps> = props => {
+  return {
+    ...props,
+    type:
+      props.type === 'muted' ? 'subtle' : props.type === 'error' ? 'danger' : props.type,
+  };
+};
+
+interface ChonkAlertProps extends Omit<AlertProps, 'type'> {
+  theme: DO_NOT_USE_ChonkTheme;
+  type: 'subtle' | 'info' | 'warning' | 'success' | 'danger';
+  size?: 'sm';
+}
+
+function Container({type, theme, size, ...props}: ChonkAlertProps) {
+  const colors = makeChonkAlertTheme(type, theme);
+  const padding = makeChonkAlertPadding(size, theme);
+
+  return (
+    <div
+      {...props}
+      css={css`
```

Comment:
I guess we might want to re-think creating a `styled('div')` out of this part. For one, it would avoid calculating `css` on every render (apparently, styled(div) is better optimized) and also, I’m seeing lots of unrecognized props being forwarded to the dom in storybook:

![Screenshot 2025-02-20 at 10 01 02](https://github.com/user-attachments/assets/d14428b1-7911-466d-952e-806f9756e61a)

styled('div') would automatically remove those.

---

The initial problem we wanted to solve is calculating the `alertTheme` on the fly, because we wanted to remove it from theme itself. I think this is still the right approach, but it might need a mixture of the container + styled components.

Reply:
@JonasBa I did something similar for `Badge` - combining a container component that creates colors with a styled component. Let me know what you think:

- https://github.com/getsentry/sentry/pull/85522

Reply:
Oh, I had no idea styled takes care of caching and bleeding props, I will change it to use a styled component!